File: 164-maximum-gap/maximum-gap.java
class Solution {
    public int maximumGap(int[] nums) {
        Arrays.sort(nums);
        int maxdiff = 0 ;
        for(int i = 1 ; i < nums.length ; i++){
            int diff = nums[i] - nums[ i - 1 ];
            if(diff > maxdiff){
                maxdiff = diff;
            }
        }
        return maxdiff;

    }
}

File: 164-maximum-gap/maximum-gap.c
int maximumGap(int* nums, int numsSize) {
    int compare(const void *a, const void *b) {
        return (*(int *)a - *(int *)b);
    }
    qsort(nums, numsSize, sizeof(int), compare);
    int maxdiff = 0;

    for(int i = 1; i < numsSize; i++) {
        int diff = nums[i] - nums[i - 1];

        if(diff > maxdiff) {
            maxdiff = diff;
        }
    }

    return maxdiff;
}